shell/common/persistent_cache.cc

```diff
@@ -85,15 +85,26 @@ bool PersistentCache::Purge() {
   FML_CHECK(GetWorkerTaskRunner());
 
   std::promise<bool> removed;
-  GetWorkerTaskRunner()->PostTask(
-      [&removed, cache_directory = cache_directory_]() {
-        if (cache_directory->is_valid()) {
-          FML_LOG(INFO) << "Purge persistent cache.";
-          removed.set_value(RemoveFilesInDirectory(*cache_directory));
-        } else {
-          removed.set_value(false);
+  GetWorkerTaskRunner()->PostTask([&removed,
+                                   cache_directory = cache_directory_]() {
+    if (cache_directory->is_valid()) {
+      // Only remove files but not directories.
+      FML_LOG(INFO) << "Purge persistent cache.";
+      fml::FileVisitor recursive_cleanup = [&recursive_cleanup](
```

Comment:
Is this different from calling `VisitFilesRecursively` with a visitor that deletes the file?

Reply:
Yes, `VisitFilesRecursively` with a simple delete visitor would also remove directories. However, I can create a visitor that only deletes file but not directory. Please see my latest patch.